### PR TITLE
support alert pagination from backend

### PR DESCRIPTION
**Pull request by chenxyz707**
support alert pagination from backend, prevent  to exceed  MongoDB RAM limit

### CODE REVIEW

File: src/components/AlertList.vue [hunk 1]
```diff
@@ -6,9 +6,9 @@
       :items="alerts"
       :pagination.sync="pagination"
       :rows-per-page-items="pagination.rowsPerPageItems"
+      :total-items="pagination.totalItems"
       class="alert-table"
       :class="[ displayDensity ]"
-      :search="search"
```
Comment (satterly) [inline src/components/AlertList.vue:11]: Why did you delete "search"?

File: src/store/modules/alerts.store.ts [hunk 2]
```diff
@@ -58,10 +58,11 @@ const mutations = {
     state.isSearching = true
     state.query = query
   },
-  SET_ALERTS(state, alerts): any {
+  SET_ALERTS(state, result): any {
     state.isLoading = false
     state.isSearching = false
-    state.alerts = alerts
+    state.alerts = result.alerts
+    state.pagination.totalItems = result.total
```
Comment (satterly) [inline src/store/modules/alerts.store.ts:65]: This could be `alerts.length` then nothing else has to change.

File: src/store/modules/alerts.store.ts [hunk 3]
```diff
@@ -155,9 +156,13 @@ const actions = {
...
 
+    // support pagination
+    params.append('page', state.pagination.page)
+    params.append('page-size', state.pagination.rowsPerPage)
+
     return AlertsApi.getAlerts(params)
-      .then(({ alerts }) => commit('SET_ALERTS', alerts))
-      .catch(() => commit('RESET_LOADING'))
+    .then(result => commit('SET_ALERTS', result))
+    .catch(() => commit('RESET_LOADING'))
```
Comment (satterly) [inline src/store/modules/alerts.store.ts:165]: If you use `alerts.length` above this can be left as it was.

**Comment by satterly** [comment]
Thanks for the PR. Can you address the comments I made?

Also, please make sure tests pass -- especially lint errors. 👍

File: src/components/AlertList.vue (hunk 1 above)
Comment (chenxyz707) [reply src/components/AlertList.vue:11]: Remove "search" makes pagination work, maybe it's a bug for vuetify data-table.
https://github.com/vuetifyjs/vuetify/issues/5419

File: src/store/modules/alerts.store.ts (hunk 2 above)
Comment (chenxyz707) [reply src/store/modules/alerts.store.ts:65]: alerts.length is the count in current page, but result.total is total count of alerts.

**Comment by chenxyz707** [comment]
@satterly Sorry, I'm a Java engineer, just know a little about front-end build tools, I will do my best to solve tests.

**Comment by satterly** [comment]
This changes the way search, filtering and paging works and so is a significant UX change when the actual issue only affects a tiny proportion of users (ie. only you?) and also where there is an existing solution for this which is to set the `DEFAULT_PAGE_SIZE` for the API. Closing.
